src/pages/add-test-center.page.js

```diff
@@ -0,0 +1,382 @@
+    snackbarSeverity: '',
+    snackbarMessage: '',
+  };
+
+  constructor(props) {
+    super(props);
+    bindAll(this, ['onCheckboxSelected', 'onCancelClicked', 'handleSubmit', 'handleSnackbarClose']);
+    this.typesService = TypesService.getInstance();
+    this.facilitateService = FacilitateService.getInstance();
+    this.state = cloneDeep(this.initialState);
+  }
+
+  async componentDidMount() {
+    this.testCenterTypes = await this.typesService.getFacilities();
+  }
+
+  onCheckboxSelected(selected) {
+    selected.value = !selected.value;
+
+    if (selected.key === 'ii' || selected.key === 'rp') {
+      if (selected.value) {
+        // add to testTypes array
+        this.setState((prevState) => ({
+          postData: {
+            ...prevState.postData,
+            testTypes: [...prevState.postData.testTypes, { id: selected.key }],
+          },
+        }));
+      } else {
+        // remove from testTypes array
+        this.setState((prevState) => ({
+          postData: {
+            ...prevState.postData,
+            testTypes: prevState.postData.testTypes.filter((key) => key.id !== selected.key),
+          },
+        }));
+      }
+    } else {
+      this.setState((prevState) => ({
+        postData: { ...prevState.postData, [selected.key]: selected.value },
+      }));
+    }
+  }
+
+  onCancelClicked() {
+    this.setState(cloneDeep(this.initialState));
+  }
+
+  async handleSubmit(e) {
+    e.preventDefault();
+
+    if (!this.state.postData.name || !this.state.postData.address) {
+      this.setState({
+        snackbarMessage: 'Please complete required fields',
+        snackbarSeverity: 'error',
+        snackbarOpen: true,
+      });
+    } else {
+      const response = await this.facilitateService.addFacilityByCitizen({ value: this.state.postData });
+      if (response.error) {
+        this.setState({
+          snackbarMessage: 'An error occured. Please try again later.',
+          snackbarSeverity: 'error',
+          snackbarOpen: true,
+        });
+      } else {
+        this.setState({
+          ...cloneDeep(this.initialState),
+          snackbarMessage: 'Success! New Test Center has been submitted',
+          snackbarSeverity: 'success',
+          snackbarOpen: true,
+        });
+      }
+    }
+  }
+
+  handleSnackbarClose() {
+    this.setState({ snackbarOpen: false });
+  }
+
+  render() {
+    return (
+      <>
+        <Section className="add-test-center-page">
+          <StyledHeader>
+            <h1>Submit New Test Center</h1>
+            <h2>
+              Complete the following form to propose a new test center within AllClear. All submissions will be reviewed
+              by the AllClear team within 24 hours.
+            </h2>
+          </StyledHeader>
+          <Container maxWidth="sm">
+            <form noValidate autoComplete="off" onSubmit={this.handleSubmit}>
+              <Article>
+                <H3>Test Center Profile</H3>
+                <TextField
```

Comment:
these could probably be a .map() from a constants array. not needed for this PR tho

Reply:
ok perfect I can fix that in a separate ticket